src/schema/v2/HeroUnit/deleteHeroUnitMutation.ts

```diff
@@ -0,0 +1,77 @@
+import { GraphQLString, GraphQLUnionType, GraphQLObjectType } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { GraphQLNonNull } from "graphql"
+import { ResolverContext } from "types/graphql"
+import { HeroUnitType } from "./HeroUnitType"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+
+interface Input {
+  id: string
+}
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "deleteHeroUnitSuccess",
+  isTypeOf: (data) => data.id,
+  fields: () => ({
+    heroUnit: {
+      type: HeroUnitType,
+      resolve: (heroUnit) => heroUnit,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "deleteHeroUnitFailure",
+  isTypeOf: (data) => data._type === "GravityMutationError",
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "deleteHeroUnitResponseOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const deleteHeroUnitMutation = mutationWithClientMutationId<
+  Input,
+  any | null,
+  ResolverContext
+>({
+  name: "deleteHeroUnitMutation",
+  description: "deletes a hero unit.",
+  inputFields: {
+    id: { type: new GraphQLNonNull(GraphQLString) },
+  },
+  outputFields: {
+    heroUnitOrError: {
+      type: ResponseOrErrorType,
+      description: "On success: the deleted hero unit.",
+      resolve: (result) => result,
+    },
+  },
```

Comment:
You may want to include the viewer type on these mutations. After a mutation generally you'll want to re-fetch the parent context to update the UI.

Reply:
Yea, sometimes its easier to wait until you're working on the UI so you know what data you'll want to re-request as a result of the mutation to make sure the UI is updated. sometimes _just_ the object in question is sufficient, but sometimes the parent, etc.

Reply:
Ooooo, I wonder if this is why I'm seeing something strange: when you create a new hero unit in Forque then you land back on the list page. On that page you get a message that the hero unit was created but the list does not update to show it. Could that be related to this? 🤔

Reply:
Yep definitely; you need to re-request the parent to update it.